### PR TITLE
seccomp: fix issue in system call filter with errno

### LINKED ISSUE #18916: SystemCallFilter doesn't override errno
**systemd version the issue has been seen with**
247.3-1

**Used distribution**
Debian Unstable

**Linux kernel version used** (`uname -a`)
`5.10.0-4-amd64 #1 SMP Debian 5.10.19-1 (2021-03-02) x86_64 GNU/Linux`

**Expected behaviour you didn't see**

The documentation of [`SystemCallFilter`](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#SystemCallFilter=) states:

> Deny-listed system calls and system call groups may optionally be suffixed with a colon (":") and "errno" error number (between 0 and 4095) or errno name such as EPERM, EACCES or EUCLEAN (see errno(3) for a full list). This value will be returned when a deny-listed system call is triggered, instead of terminating the processes immediately.

This only seems to work on the first directive in which the call is denied; any further overrides will be unable to change the error number.

**Steps to reproduce the problem**

```
[Service]
ExecStart=/bin/uname
SystemCallErrorNumber=EUCLEAN
SystemCallFilter=@system-service
SystemCallFilter=~uname:EBADF
```

When attempting to run this service, the `uname` error (as reported in the journal) is "Structure needs cleaning", i.e. `EUCLEAN`. I would have expected to see `EBADF` instead.

As far as I can tell this makes it impossible to customize errno on a per-system-call basis when using an allow-list approach.

### PR DESCRIPTION
**Pull request by yuwata**
Fixes #18916.

### CODE REVIEW

File: src/shared/seccomp-util.c [hunk 1]
```diff
@@ -1063,7 +1063,7 @@ int seccomp_load_syscall_filter_set_raw(uint32_t default_action, Hashmap* set, u
         uint32_t arch;
         int r;
 
-        /* Similar to seccomp_load_syscall_filter_set(), but takes a raw Set* of syscalls, instead of a
+        /* Similar to seccomp_load_syscall_filter_set(), but takes a raw Hashmap* of syscalls, instead of a
```
Comment (keszybz) [inline src/shared/seccomp-util.c]: Hmm, maybe also rename `set` to `map` or `filter` or something? The variable name is misleading too…

File: src/core/load-fragment.c [hunk 2]
```diff
@@ -3218,6 +3218,10 @@ int config_parse_syscall_filter(
                         log_syntax(unit, LOG_WARNING, filename, line, r, "Failed to parse syscall:errno, ignoring: %s", word);
                         continue;
                 }
+                if (!invert && num >= 0) {
+                        log_syntax(unit, LOG_WARNING, filename, line, r, "Allow listed system calls cannot take error number, ignoring: %s", word);
```
Comment (keszybz) [inline src/core/load-fragment.c]: "Allow-listed"

**Review by keszybz — COMMENTED** [review]
LGTM. Minor suggestions only.

**Comment by yuwata** [comment]
@keszybz Thank you for your review. The above two points are addressed. Upgrading the green label.

**Comment by poettering** [comment]
lgtm

**Comment by topimiettinen** [comment]
lgtm too.

**Pull request merged**